### PR TITLE
NeedlePlot app redesign

### LINKED ISSUE #225: NeedlePlot redesign
Update design of NeedlePlot app in accordance with [this streambed card](https://github.com/plotly/streambed/projects/16#card-18258548).

### PR DESCRIPTION
**Pull request by shammamah**
Closes #225 

## About 
- [ ] This is a new component 
- [ ] I am adding a feature to an existing component, or improving an existing feature
- [x] I am closing an issue 

## Description of changes 
* Change colors 
* Move controls to "About", "Data", and "Graph" tabs 
* Make graph larger

## Before merging
- [x] I have gone through the [code review checklist](https://github.com/plotly/dash-component-boilerplate/blob/master/%7B%7Bcookiecutter.project_shortname%7D%7D/review_checklist.md)
- [x] I have completed all of the [steps to take before merging](https://github.com/plotly/dash-bio/blob/master/.github/before_merging.md)
- [x] I know how to [deploy to DDS](https://github.com/plotly/dash-bio/blob/master/.github/deploying_to_dds.md)

### CODE REVIEW

**Comment by shammamah** [comment]
Oops -- it's actually not quite ready for review yet.

File: tests/dashbio_demos/app_needle_plot.py [hunk 1]
```diff
@@ -103,108 +103,258 @@ def description():
...
+            dcc.Tabs(id='needleplot-tabs', children=[
+                dcc.Tab(
+                    label='About',
+                    value='what-is',
+                    children=html.Div(className='needleplot-tab', children=[
+                        html.H4('What is Needle Plot?'),
+                        html.P('Needle Plot allows you to display mutations in '
+                               'a genome. Due to its similarity to both a barplot '
+                               'and a scatter plot, it can be used to plot '
+                               'datasets that have too many categories for a '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]: Asking out of ignorance: is 'category' the bioinformatics term? Does 'everyone' know what it refers to, in the context of genomic data?

**Comment by mkcor** [comment]
Maybe this is outside the scope of this PR, but it would make sense to disable (grey out?) the "Dataset:" dropdown when Dataset source is *not* demo dataset:
upload_data

File: tests/dashbio_demos/app_needle_plot.py [hunk 2]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                    value=0
+                                ),
+                            ]
+                        ),
+
+                        html.Hr(),
+                        html.Div(
+                            id='needle-protein-domains-select-div',
+                            className='needle-config-data',
+                            title='If checked, it will allow the user to load '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]: You have to decide between 'you' and 'the user' and it looks like we're going for 'you.'
So I would reword in a much more direct way: "Check to load mutation data ..."

File: tests/dashbio_demos/app_needle_plot.py [hunk 3]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                            ]
+                        ),
+
+                        html.Div(
+                            id='needle-%s-div' % FILE_KEY,
+                            className='needle-load-option-div',
+                            children=[
+                                html.Div(
+                                    id='needle-mutdata-file-div',
+                                    title='Mutation data files are json files containing '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                    title='Mutation data files are JSON files containing '
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 4]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'the fields :\n'
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                          'the following fields :\n'
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 5]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          '- "x" (protein coordinate of the mutation) \n'
+                                          '- "y" (number of recorded mutations) \n'
+                                          '- "mutationGroups" (type of mutations) \n'
+                                          '- "domains" (protein domains.\n '
+                                          '"x", "y", "mutationGroups" are arrays, they '
+                                          'must have the same length or be empty, '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                          'must have the same length, '
```
If they are empty, their length is zero, so it doesn't add information, does it?

File: tests/dashbio_demos/app_needle_plot.py [hunk 6]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          '"x" is required. The "domains" is an array of '
+                                          'json objects with the required fields "name" '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                          'JSON objects with required fields "name" '
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 7]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'and "coord", which are a string and a string '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]: This is very unclear... Are both "name" and "coord" strings?

File: tests/dashbio_demos/app_needle_plot.py [hunk 8]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'detailing the start and end of the domains in '
+                                          'protein coordinate (eg. "23-34"), '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                          'protein coordinate (e.g., "23-34"), '
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 9]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'respectively.',
+                                    children=[
+                                        html.H5(
+                                            'Upload mutation data json file'
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                            'Upload mutation data JSON file'
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 10]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                        )
+                                    ]
+                                ),
+                                html.Div(
+                                    id='needle-domain-file-div',
+                                    title='Protein data files accepted here can be two'
+                                          ' fold : \n'
+                                          '- an array of json objects with the required '
+                                          'fields "name" and "coord", which are a string '
+                                          'and a string  detailing the start and end of '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]: Same comment / question as above.

File: tests/dashbio_demos/app_needle_plot.py [hunk 11]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'the domains in protein coordinate (eg. "23-34")'
+                                          ', respectively.\n'
+                                          '- an json file with the same structure as the '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                          '- a JSON file with the same structure as '
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 12]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'one in the PFAM database : '
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]: and I would remove this line

File: tests/dashbio_demos/app_needle_plot.py [hunk 13]
```diff
@@ -103,108 +103,258 @@ def description():
...
+                                          'http://pfam.xfam.org/protein/P04637/graphic.',
+                                    children=[
+                                        html.H5('Upload protein data json file'),
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
                                        html.H5('Upload protein data JSON file'),
```

File: tests/dashbio_demos/app_needle_plot.py [hunk 14]
```diff
@@ -103,108 +103,258 @@ def description():
...
-                                                ],
-                                                values=[]
-                                            )
+                                            id='needle-mutdata-file-info-div'
+                                        )
+                                    ]
+                                ),
+                                html.Div(
+                                    id='needle-domain-file-div',
+                                    title='Protein data files accepted here can be two'
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]: I guess you mean "can be of two types:"

File: tests/dashbio_demos/app_needle_plot.py (hunk 1 above)
Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: Needle plots in general follow this definition (http://support.sas.com/documentation/cdl/en/bidbrdug/67143/HTML/default/viewer.htm#n078fn4uh7lvdtn1p9c0hrindwfl.htm). We could add in a more bioinformatics-specific word. However, I think that datasets (e.g., number of mutations for different locations in the gene) that would be used with the needle plot will be this way by nature.

File: tests/dashbio_demos/app_needle_plot.py (hunk 5 above)
Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: I think @Bachibouzouk meant that one of them can be empty, and the other of arbitrary length?

File: tests/dashbio_demos/app_needle_plot.py (hunk 7 above)
Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: @Bachibouzouk can answer this one :)

File: tests/dashbio_demos/app_needle_plot.py (hunk 12 above)
Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: It might be important to have an example of the structure the JSON file should have, but maybe we should instead link to one of the files we have for example data in this repo.

File: tests/dashbio_demos/app_needle_plot.py (hunk 2 above)
Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: Changed in d81a6a0!

File: tests/dashbio_demos/app_needle_plot.py (hunk 14 above)
Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: Fixed in d81a6a0!

File: tests/dashbio_demos/app_needle_plot.py (hunk 12 above)
Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: Absolutely! My suggestion is only about the wording: "... same structure as <link>"

File: tests/dashbio_demos/app_needle_plot.py (hunk 10 above)
Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: I don't understand if "23-34" is a value example for "name" or "coord" (or both?).

File: tests/dashbio_demos/app_needle_plot.py (hunk 1 above)
Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: I'm only talking about the term 'category' (as present in the data, not the plot). Isn't it... "too many mutations?"

Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: I would reword along the lines of "Combining barplot characteristics with scatter plot characteristics, needle plots are best suited to datasets which have too many mutations(?) for a barplot to be meaningful."

**Comment by shammamah** [comment]
@mkcor Made most of the changes you suggested. I think that the other suggestions require some input from @Bachibouzouk, so let's get this one merged in and then amend the remaining errors in a future PR.

File: tests/dashbio_demos/app_needle_plot.py [hunk 15]
```diff
@@ -103,108 +103,260 @@ def description():
 
 def header_colors():
     return {
-        'bg_color': '#009900',
-        'font_color': '#DFE8F3',
-        'light_logo': False
+        'bg_color': '#0D1A51',
+        'font_color': '#FFF',
```
Comment (mkcor) [inline tests/dashbio_demos/app_needle_plot.py]:
```suggestion
        'font_color': '#FFFFFF',
```

Comment (shammamah) [reply tests/dashbio_demos/app_needle_plot.py]: Don't those two mean the same thing?

Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: Yes, in a minimalistic perspective, I can understand why you'd keep only 3 F's, since they're enough (to work). But, in a standard-oriented perspective, I don't know why we wouldn't go for the standard hexadecimal notation...

File: tests/dashbio_demos/app_needle_plot.py (hunk 7 above)
Comment (Bachibouzouk) [reply tests/dashbio_demos/app_needle_plot.py]: Yes they are both strings

Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: If we talk about two fields ("name" and "coord") but then give just one example, there's no way to guess if it's an example value for "name" or for "coord"?!?
Continuing the conversation over at #277.

File: tests/dashbio_demos/app_needle_plot.py (hunk 10 above)
Comment (mkcor) [reply tests/dashbio_demos/app_needle_plot.py]: Resolving in #293.
